enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,335 @@
+
+`cluster-bootstrap` will apply all the required manifests (under ``/opt/openshift/manifests/``)
+
+Bootkube will get the master Ignition from `machine-config-server` and
+generate an updated master Ignition combining the original ignition with the control plane static pods manifests and all required resources including etcd data.
+`bootkube.sh` will write the new master Ignition along with RHCOS to disk.
+At this point `bootkube` will reboot the node and let it complete the cluster creation.
+
+After the host reboots, the `kubelet` service will start the control plane static pods.
+Kubelet will send a CSR (see below) and join the cluster.
+CVO will deploy all cluster operators.
+The control plane operators will rollout a new revision (if necessary).
+
+#### OpenShift-installer
+
+We will add logic to the installer to create `bootstrap-in-place-for-live-iso.ign` Ignition config.
+This Ignition config will diverge from the default bootstrap Ignition:
+bootkube.sh:
+1. Start cluster-bootstrap without required pods (`--required-pods=''`)
+2. Run cluster-bootstrap with `--bootstrap-in-place` entrypoint to enrich the master Ignition.
+3. Write RHCOS image and the master Ignition to disk.
+4. Reboot the node.
+
+#### Cluster-bootstrap
+
+By default, `cluster-bootstrap` starts the bootstrap control plane and creates all the manifests under ``/opt/openshift/manifests``.
+`cluster-bootstrap` also waits for a list of required pods to be ready. These pods are expected to start running on the control plane nodes.
+In case we are running the bootstrap in place, there is no control plane node that can run those pods. `cluster-bootstrap` should apply the manifest and tear down the control plane. If `cluster-bootstrap` fails to apply some of the manifests, it should return an error.
+
+
+`cluster-bootstrap` will have a new entrypoint `--bootstrap-in-place` which will get the master Ignition as input and will enrich the master Ignition with control plane static pods manifests and all required resources including etcd data.
+
+#### Bootstrap / Control plane static pods
+
+The control plane components we will add to the master Ignition are (to be placed under `/etc/kubernetes/manifests`):
+
+1. etcd-pod
+2. kube-apiserver-pod
+3. kube-controller-manager-pod
+4. kube-scheduler-pod
+
+Control plane required resources to be added to the Ignition:
+
+1. `/var/lib/etcd`
+2. `/etc/kubernetes/bootstrap-configs`
+3. /opt/openshift/tls/* (`/etc/kubernetes/bootstrap-secrets`)
+4. /opt/openshift/auth/kubeconfig-loopback (`/etc/kubernetes/bootstrap-secrets/kubeconfig`)
+
+**Note**: `/etc/kubernetes/bootstrap-secrets` and `/etc/kubernetes/bootstrap-configs` will be deleted after the node reboots by the post-reboot service (see below), and the OCP control plane is ready.
+
+The control plane operators (that will run on the node post reboot) will manage the rollout of a new revision of the control plane pods.
+
+#### etcd data
```

Comment:
There are 2 options listed in this section. Do we know which is preferred, yet?

Reply:
Yes, according to @hexfusion the second option (stop the pod and copy all the data) is better.
Updating